Scripts/Utilities/ComponentUtils.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace UnityEditor.Experimental.EditorVR.Utilities
+{
+	/// <summary>
+	/// Special utility class for getting components in the editor without allocations
+	/// </summary>
+	/// <typeparam name="T"></typeparam>
+	public class ComponentUtils<T>
+	{
+		static List<T> s_RetrievalList = new List<T>();
+
+		public static T GetComponent(GameObject go)
```

Comment:
Why not an extension method? and a static class?

Reply:
Actually, maybe you can't do the static class because of the templting. And a static method would no doubt need a different name. I can check and see if it's possible.

Reply:
Yes - this class and method has to stay because of the templating.  We could wrap it in an extension method (as we are explicitly calling out what object/interface we are using), but I'm not convinced we want to.  Unity already has methods for getting a component from a gameobject and I don't want to confuse users by adding duplicate methods on their intellisense - especially since this is for an EVR specific optimization.

Additionally, our ObjectUtils.AddComponent functions are also not extension methods.

Reply:
Just tried the extension method, and indeed we can't put extension methods in templated classes. However, we can use a templated static class, which makes it clear that you should never instantiate ComponentUtils. I would still say we should make ComponentUtils static.